implementation/src/main/java/io/smallrye/mutiny/operators/multi/MultiOnSubscribeCall.java

```diff
@@ -63,6 +66,61 @@ public void onSubscribe(Flow.Subscription s) {
                 s.cancel();
             }
         }
+
+        @Override
+        public void onFailure(Throwable throwable) {
+            lock.lock();
+            if (!uniHasTerminated) {
+                terminatedEarly = true;
+                this.failure = throwable;
+                lock.unlock();
+            } else {
+                lock.unlock();
+                super.onFailure(throwable);
```

Comment:
I would add a comment explaining why you cannot use a try/finally because you need to invoke the callback without the lock.